lta/nersc_verifier.py

```diff
@@ -0,0 +1,240 @@
+        self.file_catalog_rest_url = config["FILE_CATALOG_REST_URL"]
+        self.tape_bath_path = config["TAPE_BASE_PATH"]
+        self.work_retries = int(config["WORK_RETRIES"])
+        self.work_timeout_seconds = float(config["WORK_TIMEOUT_SECONDS"])
+
+    def _do_status(self) -> Dict[str, Any]:
+        """NerscVerifier has no additional status to contribute."""
+        return {}
+
+    def _expected_config(self) -> Dict[str, Optional[str]]:
+        """NerscVerifier provides our expected configuration dictionary."""
+        return EXPECTED_CONFIG
+
+    async def _do_work(self) -> None:
+        """Perform a work cycle for this component."""
+        self.logger.info("Starting work on Bundles.")
+        work_claimed = True
+        while work_claimed:
+            work_claimed = await self._do_work_claim()
+        self.logger.info("Ending work on Bundles.")
+
+    async def _do_work_claim(self) -> bool:
+        """Claim a bundle and perform work on it."""
+        # 1. Ask the LTA DB for the next Bundle to be verified
+        # configure a RestClient to talk to the LTA DB
+        lta_rc = RestClient(self.lta_rest_url,
+                            token=self.lta_rest_token,
+                            timeout=self.work_timeout_seconds,
+                            retries=self.work_retries)
+        self.logger.info("Asking the LTA DB for a Bundle to verify at NERSC with HPSS.")
+        pop_body = {
+            "claimant": f"{self.name}-{self.instance_uuid}"
+        }
+        response = await lta_rc.request('POST', '/Bundles/actions/pop?dest=NERSC&status=verifying', pop_body)
+        self.logger.info(f"LTA DB responded with: {response}")
+        bundle = response["bundle"]
+        if not bundle:
+            self.logger.info("LTA DB did not provide a Bundle to verify at NERSC with HPSS. Going on vacation.")
+            return False
+        # process the Bundle that we were given
+        if await self._verify_bundle_in_hpss(lta_rc, bundle):
+            await self._add_bundle_to_file_catalog(bundle)
+            await self._update_bundle_in_lta_db(lta_rc, bundle)
+        return True
+
+    async def _add_bundle_to_file_catalog(self, bundle: BundleType) -> bool:
+        """Add a FileCatalog entry for the bundle, then update existing records."""
+        # configure a RestClient to talk to the File Catalog
+        fc_rc = RestClient(self.file_catalog_rest_url,
+                           token=self.file_catalog_rest_token,
+                           timeout=self.work_timeout_seconds,
+                           retries=self.work_retries)
+        # determine the path where the bundle is stored on hpss
+        data_warehouse_path = bundle["path"]
+        basename = os.path.basename(bundle["bundle_path"])
+        stupid_python_path = os.path.sep.join([self.tape_bath_path, data_warehouse_path, basename])
+        hpss_path = os.path.normpath(stupid_python_path)
```

Comment:
We may want to stick something like `tape:` or `archive:` onto the front of `hpss_path`, to make it clear it's not a normal file available for download.

Reply:
We've had the convention of treating the `logical_name` field like a path. I think adding a tag like that in the path itself will lead to some messy conditional code for handling the data. If we want a tag like that, we could make it another field on the record itself. Maybe `"hpss": true,` ?

Reply:
Maybe just `"online": false` to signal that this filesystem is not for online access?

Reply:
This is very similar to the archive flag in meaning though.  This is a tossup as to what to use.

Reply:
I put both `"hpss": true,` and `"online": false,` into the location for the archive bundle.